standard/lexical-structure.md

```diff
@@ -974,56 +1058,63 @@ A `#undef` may "undefine" a conditional compilation symbol that is not defined.
 The conditional compilation directives are used to conditionally include or exclude portions of a compilation unit.
 
 ```ANTLR
-Pp_Conditional
-    : Pp_If_Section Pp_Elif_Section* Pp_Else_Section? Pp_Endif
+fragment PP_Conditional
+    : PP_If_Section
+    | PP_Elif_Section
+    | PP_Else_Section
+    | PP_Endif
     ;
 
-Pp_If_Section
-    : Whitespace? '#' Whitespace? 'if' Whitespace Pp_Expression Pp_New_Line Conditional_Section?
-    ;
-    
-Pp_Elif_Section
-    : Whitespace? '#' Whitespace? 'elif' Whitespace Pp_Expression Pp_New_Line Conditional_Section?
+fragment PP_If_Section
+    : 'if' PP_Whitespace PP_Expression
     ;
     
-Pp_Else_Section
-    : Whitespace? '#' Whitespace? 'else' Pp_New_Line Conditional_Section?
+fragment PP_Elif_Section
+    : 'elif' PP_Whitespace PP_Expression
     ;
     
-Pp_Endif
-    : Whitespace? '#' Whitespace? 'endif' Pp_New_Line
+fragment PP_Else_Section
+    : 'else'
     ;
     
-Conditional_Section
-    : Input_Section
-    | Skipped_Section_Part+
+fragment PP_Endif
+    : 'endif'
     ;
+```
 
-Skipped_Section_Part
-    : Skipped_Characters? New_Line
-    | Pp_Directive
-    ;
-    
-Skipped_Characters
-    : Whitespace? Not_Number_Sign Input_Character*
-    ;
+Conditional compilation directives shall be written as sets consisting of, in order, a `#if` directive, zero or more `#elif` directives, zero or one `#else` directive, and a `#endif` directive. Between the directives are ***conditional sections*** of source code. Each section is controlled by the immediately preceding directive. A conditional section may itself contain nested conditional compilation directives provided these directives form complete sets.
```

Comment:
This is where we pay the price of not being able to sequence the conditional directives in grammar. That's fine - the term "set" may be confusing here, because it does not refer to a mathematical set (unordered by definition). Maybe another term, like "group", would be better? (Yes, groups are also a mathematical notion, but less likely to be understood as such in this context!)

Reply:
I think "group" would fit.

Reply:
The group agreed with group.

Reply:
(Group-think in action.)